### PR TITLE
Updated Azure.APIM.EncryptValues

### LINKED ISSUE #2146: Encrypt all API Management named values with Key Vault secrets
### Existing rule

Azure.APIM.EncryptValues

### Suggested rule

This rule is currently checking for encryted strings (custom secrets), where referencing secrets in Azure Key Vault may be preferred instead as it helps improve API Management security.

Creating a new rule for this can cause conflicts for customers, because after they have fixed `Azure.APIM.EncryptValues` the new rule will cause conflict. Then after configuring the new rule, `Azure.APIM.EncryptValues` is back to failing.

We should instead refactor current rule and advise customers to use Key Vault secrets which helps improve API Management security.

Additionally it might be overkill to check for plain text, but if we only targeted against secrets, we would need two rules so the purpose is clear, open for feedback and suggestions here.

### Pillar

Security

### Additional context

- [Use named values in Azure API Management policies](https://learn.microsoft.com/azure/api-management/api-management-howto-properties)
- [Prerequisites for key vault integration](https://learn.microsoft.com/azure/api-management/api-management-howto-properties?tabs=azure-portal#prerequisites-for-key-vault-integration)
- [Azure deployment reference](https://learn.microsoft.com/azure/templates/microsoft.apimanagement/service/namedvalues#keyvaultcontractcreatepropertiesorkeyvaultcontractpr)

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Updated Azure.APIM.EncryptValues rule.

Breaking change, unsure if baseline should be updated.

Fixes #2146

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [ ] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
@BernieWhite Let me know :)

File: src/PSRule.Rules.Azure/rules/Azure.APIM.Rule.ps1 [hunk 1]
```diff
@@ -333,7 +336,9 @@ function global:GetAPIMPolicyNode {
     [CmdletBinding()]
     param (
         [Parameter(Mandatory)]
-        [string]$Node
+        [string]$Node,
+        [Parameter()]
+        [switch]$GlobalPolicy
```
Comment (BernieWhite) [inline src/PSRule.Rules.Azure/rules/Azure.APIM.Rule.ps1:341]:
```suggestion
        [string]$Node
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
@BenjaminEngeset looks great. Just a few minor tweak and it should be good to go. When you update the `ruleSet` it will reduce the number of older baselines so you will need to subtract one from each of these numbers in the tests for this to pass.

**Comment by BenjaminEngeset** [comment]
> @BenjaminEngeset looks great. Just a few minor tweak and it should be good to go. When you update the `ruleSet` it will reduce the number of older baselines so you will need to subtract one from each of these numbers in the tests for this to pass.

Hey Bernie, do you mind giving me the instruction for this? I'm unsure where this should be changed. Thanks.

**Review by BernieWhite — COMMENTED** [review]
Hi @BenjaminEngeset I'm made the updates to the tests and a few changes to the docs.

Please review and let me know if you are happy with the updates from your perspective, or if you want to make any further changes. Otherwise we should be good to merge.

**Comment by BenjaminEngeset** [comment]
> Hi @BenjaminEngeset I'm made the updates to the tests and a few changes to the docs.
> 
> Please review and let me know if you are happy with the updates from your perspective, or if you want to make any further changes. Otherwise we should be good to merge.

@BernieWhite Was just checking, I am very happy and great updates making it even better. Lets merge.

**Review by BernieWhite — APPROVED** [review]
Thanks @BenjaminEngeset. All good to merge.

**Pull request merged**